napari/layers/labels/labels.py

```diff
@@ -575,6 +585,37 @@ def color(self, color):
 
         self.color_mode = color_mode
 
+    def _validate_colors(self, labels: Iterable[int]):
+        """Check whether any of the given labels will be aliased together.
+
+        See https://github.com/napari/napari/issues/6084 for details.
+        """
+        aliases: Dict[float, List[int]] = defaultdict(list)
+        labels = [k for k in labels if k is not None]
+        labels_int = np.fromiter(labels, dtype=int)
+        labels_texture = self._to_vispy_texture_dtype(labels_int)
+        for integer, texture in zip(labels_int, labels_texture):
+            aliases[texture].append(integer)
+        aliased_list = [v for k, v in aliases.items() if len(v) > 1]
```

Comment:
Many python loops and list comprehensions here, hopefully not a problem with a lot of labels?

Reply:
Rewrite using numpy unique. 
Cut from 8.47s to 408ms when providing colors without collisions for 10000000 entries.